### PR TITLE
update identity gems and get tests to pass

### PR DESCRIPTION
**Pull request by jgrevich**
This updates the identity client gems and code needed to keep tests passing. This also updates the CircleCI config to use Ruby 3.3.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
   gem 'rspec'
 end
 
 group :development, :test do
-  gem 'aamva', github: '18F/identity-aamva-api-client-gem', tag: 'v4.1.0'
-  gem 'identity-doc-auth', github: '18F/identity-doc-auth', tag: 'v0.4.1'
-  gem 'lexisnexis', github: '18F/identity-lexisnexis-api-client-gem', tag: 'v3.1.0'
+  gem 'aamva', github: '18F/identity-aamva-api-client-gem', tag: 'v4.2.0'
+  gem 'identity-doc-auth', github: '18F/identity-doc-auth', tag: 'v0.13.0'
+  gem 'lexisnexis', github: '18F/identity-lexisnexis-api-client-gem', tag: 'v3.2.1'
```
Comment (zachmargolis) [inline Gemfile:23]: I'd just remove these entirely since we're no longer using them

**Review by zachmargolis — APPROVED** [review]
LGTM

File: Gemfile (hunk 1 above)
Comment (jgrevich) [reply Gemfile:23]: Will do.

File: spec/usps_spec.rb [hunk 2]
```diff
@@ -1,124 +0,0 @@
-require 'spec_helper'
-require 'faraday'
-
-RSpec.describe 'USPS IPPaaS' do
```
Comment (zachmargolis) [inline spec/usps_spec.rb:4]: this spec might still be worth keeping

**Pull request merged**